### PR TITLE
Add Popout Grid to Timestamp

### LINKED ISSUE #1289: Change position of Timestamp wrapper for GEL size classes
**Is your feature request related to a problem? Please describe.**
Following #1280.

UX have updated the designs for the timestamp components.

**Describe the solution you'd like**
The element grouping both the "published" & "updated"  (likely the `div` in the Timestamp container) should become a Simorgh-only component in it's own right, containing CSS rules to handle positioning based on GEL viewport sizes.

There are two scenarios:
1008px

- At 1008px (GEL Group 4) & below, the timestamps are positioned directly above the next element, left-aligned.

1280px

- At 1280px (GEL Group 5) & above, the timestamps are positioned _alongside_ the next component. 

Screenshot of layout with grid dev tools overlayed:
Screen Shot timestamp with grid overlay

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Snapshot tests sufficient until we have visual regression screenshot tests

**Additional context**
It's possible this kind of "drop into a specific left/right column, depending on LTR/RTL" behaviour, will be a common behaviour as we add onward journeys & similar. It might be worth exploring how we could abstract this in a generic & reusable way.

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by 12**
Resolves #1289 

**Overall change:** Make the timestamp appear alongside the article content when in the GEL group 5 breakpoint

Behaviour at group 1 - 4

Behaviour at Group 5 breakpoint

**Code changes:**

- Add new CSS to allow the Timestamp to pop out to the left
- update tests

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

**Review by oluoluoxenfree — APPROVED** [review]
LGTM locally :)

**Comment by dr3** [comment]
Youre lucky the timestamp font is small, any bigger text and this change would expand the size of the paragraph next to it. If you lot add any more info into the side column with the timestamp this will be a problem 🙈 
image

**Comment by 12** [comment]
We’ve accounted for that

File: src/app/lib/layoutGrid.js [hunk 1]
```diff
@@ -303,3 +303,28 @@ export const gridContainerSmallCss = css`
...
+  }
+  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
+    grid-column: 3 / span 5;
+    max-width: ${fiveOfSixColumnsMaxWidthGroup4};
+  }
+  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
+    grid-column: ${props => props.gridColumnStart} / span
+      ${props => specifiedOrMaximum(props.gridSpan, 22)};
+    max-width: ${tenOfTwelveColumnsMaxWidthGroup5};
+    max-height: 1.375rem;
```
Comment (twinlensreflex) [inline src/app/lib/layoutGrid.js]:
```suggestion
    max-height: 1.15rem;
```
I think we should change this value. I went to take a screenshot in storybook of the behaviour that Drew and Rhys mentioned and you can see there's a little gap where the grid item has popped out.
Current grid pop out behaviour in storybook

But if you change the `max-height` to 0 we get this instead:
Possible future pop out grid item behaviour

Changing `max-height` to `1.15rem` has the same effect as 0.
Thank goodness for those coloured blocks?

**Review by jroebu14 — APPROVED** [review]
LGTM

File: src/app/lib/layoutGrid.js [hunk 2]
```diff
@@ -303,3 +303,28 @@ export const gridContainerSmallCss = css`
...
+    grid-column: 1 / span 5;
+    max-width: ${fiveOfSixColumnsMaxWidthScaleable};
+  }
+  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
+    grid-column: 3 / span 5;
+    max-width: ${fiveOfSixColumnsMaxWidthGroup4};
+  }
+  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
+    grid-column: ${props => props.gridColumnStart} / span
+      ${props => specifiedOrMaximum(props.gridSpan, 22)};
```
Comment (twinlensreflex) [inline src/app/lib/layoutGrid.js]:
```suggestion
      ${props => specifiedOrMaximum(props.gridSpan, 20)};
```

**Review by sareh — APPROVED** [review]
Looks good.
Also great work to reduce duplication of styles in the last few commits. 👍

**Comment by jamesbrumpton** [comment]
LGTM

**Comment by 12** [comment]
Following a UX review, it's been advised to try and better align the timestamp with the paragraph text, as the timestamp appears to be higher than the paragraph text.

**Comment by sareh** [comment]
@rhell Looks good for this PR. Will discuss standardisation of spacing for this pop-out example with UX when we have a second usecase for the utility.

**Pull request merged**